### DISCUSSION TITLE
Multiple screens

### DISCUSSION THREAD

**Discussion by pipikupi** [Bodmer/TFT_eSPI#2903] [General]
Hi I would like to use tft_espi lib to build a fake nixie clock. The problem is when have to work with 6 screens. [IMG_20231017_014432.jpg](https://github.com/Bodmer/TFT_eSPI/assets/76049929/12f68718-708f-488e-bec7-07c89a69f16d)

Used to work with 2 screens with switching CS pin by sketch but not sure how to deal with 6 screens.

**Comment by Bodmer** [comment]
Use the same approach with 6 CS pins, one for each display.

**Reply by pipikupi** [reply]
Hi really appreciated for very fast response. I was away for few days but did some check and it works very fine but became other issue. My sketch is very simple 1000ms loop for time to keep nice looking 6 digit clock and 10 jpg pics stored in spiffs which are displayed one on each displays and sum of the times for working with screens is around 900-1050ms which is causing displaying properly images because refreshing time sometimes is longer than time loop. Would you recommend me any solution for that case to reduce screens fefreshing time. Will be very appreciated for any advice. Thanks Pit

**Comment by pipikupi** [comment]
Hi finally have lcds arrived but I'm not able to work with more than 2 displays the same time. Was reading in the past that CS switching work just for 2 displays but not sure why

**Reply by Bodmer** [reply]
You should be able to use 6 displays. Make sure the power supply can handle the current demand for 6 displays. You may need to drop the SPI frequency due to the bus loading. Post your test sketch so I can check it. Post a wiring diagram.

**Reply by pipikupi** [reply]
Hi
Thanks a lot  for a quick response.  
Down below there is everything about my project. I'm using 5v 4A power supply for tests. Each display got its own regulator to 3.3v. Did some checks and it works very smoothly about handling demand of power. When 2 displays are connected it works very fine but with third one is stops at all. Forgive me wasn't sure about posting a code so attached all in zip file. Will be really appreciate for any sugestions  or advice



[fake_nixie_clock.zip](https://github.com/Bodmer/TFT_eSPI/files/13183519/fake_nixie_clock.zip)

**Reply by Bodmer** [reply]
40MHz bus speed may be too high for the extra bus loading when displays are added (displays typically work at his speed singly, but it is overclocking) . Try 10MHz as a test. Keep wires short to avoid "clock ringing" and reflections (these degrade the signal integrity), and make sure there is a 0V (GND) wire routed from each display directly back to the processor to minimise ground bounce. A star connection of wires would be better than a "daisy chain".

**Reply by pipikupi** [reply]
Hi Thanks for response. Made the shortest connections and so far can use up to 3 displays  but third is a bit distorted. Tested with changing the bus speed and surprisingly at 10MHz there is nothing. It works as described above at 80MHz. Changing speed from user_setup for my display. Decided to do some improvement about connections and sketch. I would like to minimise bus use and refresh just displays which is changing digit. Hopefully it will help

**Reply by Bodmer** [reply]
There's something wrong somewhere if it does not work at 10MHz. If you have an oscilloscope have a look at the SPI clock and data lines to see if they are clean and of the correct amplitude. Make sure there are plenty of GND connections straight back to the processor.

**Reply by pipikupi** [reply]
That's why I'm confused. Unfortunately haven't got an oscilloscope. But will make a new board because current one night be to everything to close and it can interfere. ![IMG_20231031_003150.jpg](https://github.com/Bodmer/TFT_eSPI/assets/76049929/d384a8e7-e06e-4d67-b9c9-188589e9c4cc)
Thank you for your support

**Reply by pipikupi** [reply]
Hi asking just in case because use another esp32 made new clear boards and it is the same spi 10MHz  nothing 40Mhz nothing 80 MHz something. So to change SPI speed I need to edit User setup line

#define SPI_FREQUENCY  40000000    

For example

#define SPI_FREQUENCY  10000000    

I know it probably is silly question

**Reply by Bodmer** [reply]
Ribbon cables do create a lot of crosstalk between the signals. I suspect this is a signal integrity problem that just happens to work with 80MHz.  You may find that 47R resistors inline with the signal at the ESP32 ( i.e GPIO -> resistor -> dislplay) will kill the reflections but you will have to reduce the speed due to the slower edge speeds. You may find the critical SPI clock line is the only one that needs this treatment and maybe the MOSI line. Seperating out the wires in the ribbon to reduce capacitance between them may also help.

**Reply by Bodmer** [reply]
You could also try reducing the drive strength of the SPI clock and MOSI lines, [see here](https://github.com/Bodmer/TFT_eSPI/discussions/2888#discussioncomment-7397596).

**Reply by pipikupi** [reply]
Thanks for advice I will test resistors inline and reducing the drive strenght for signal lines at the weekend time. About the ribbons I used to work with radio frequency converters at 144-146MHz and 430-440MHz as a hobby and during that time I've learned to use double lines ribbons and mix it like signal-ground-signal-ground etc to reduce crosstalk for lenght of ribbon up to 10cm and over that just use screened wires for signal lines. So for example for 4 SPI lines I use 10 lines ribbon
